### DISCUSSION TITLE
[General] High fees when bridging Native token (L1 -> L2)

### DISCUSSION THREAD

**Discussion by aphexyuri** [zkSync-Community-Hub/zksync-developers#734] [General] [answered]
### Team or Project

Treasure

### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

We have noticed significantly high gas fees when bridging native tokens from Sepolia to our L2 testnet (L1 ERC20 → L2 Native Token).

Below is a comparison between (left) our legacy Arbitrum Orbit testnet, and (right) or new ZK Stack testnet, both sending 1 $MAGIC from Sepolia to the respective testnets.

![Screenshot 2024-10-15 at 4 59 35 PM](https://github.com/user-attachments/assets/888daae6-ccc7-420d-b046-63130ae9d7b6)

### Additional Details

Arbitrum Orbit bridge: https://app.treasure.lol/chain/bridge
ZK Stack bridge: https://portal.topaz.treasure.lol/bridge

**Comment by alhenaai** [comment]
The high gas fees you're experiencing when bridging native tokens from L1 (Sepolia) to L2 on the ZK Stack testnet could be influenced by several factors specific to the ZKsync network and the underlying mechanisms of the bridging process.

Here are a few considerations and steps you might explore to potentially reduce these fees:

1. **Gas Price and Network Congestion**: The gas price on L1 directly affects the cost of bridging operations. Check the current gas prices and consider initiating transactions during off-peak hours.

2. **Transaction Complexity**: The complexity of the transaction can also impact the gas cost. Simplifying the transaction or optimizing the smart contract code may help reduce fees.

3. **Fee Model**: ZKsync uses a specific fee model that might differ from Arbitrum. Understanding these models can provide insights into fee components. You can review the ZKsync fee model [here](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/advanced/07_fee_model.md).

4. **Batching Transactions**: If possible, batching multiple bridging operations into a single transaction can spread the transaction costs across multiple operations, reducing the per-operation cost.

5. **Check for Updates or Patches**: Ensure that you are using the latest versions of the bridging contracts and tools, as updates may include optimizations for cost reduction.

If these steps do not resolve the issue, or if the fees remain significantly higher than expected, it may be beneficial to reach out directly to the ZKsync support or developer community for more tailored assistance or to report a potential issue with the testnet implementation.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by kaymomin** [comment]
Hey, thanks for the report, I have passed this information to the team and will investigate it.

**Comment by vasyl-ivanchuk** [comment, marked as answer]
Hey, we cannot directly compare Arbitrum bridging costs to ZKsync bridging costs since they use different technologies. Just to make sure that there is nothing wrong with fee estimation for custom base tokens on ZKsync, we checked the corresponding parts of both SDK and Portal, and the logic seems to be correct. I’ll mark this question as answered, but if you believe that the question is not resolved (e.g. depositing $MAGIC is more expensive than depositing other custom base tokens on ZK Stack), feel free to either re-open, or start a new discussion with more details.
